pkgs/stdenv/freebsd/unpack-bootstrap-files.sh

```diff
@@ -0,0 +1,56 @@
+$src/libexec/ld-elf.so.1 $src/bin/mkdir $out
+$src/libexec/ld-elf.so.1 $src/bin/tar -I "$src/libexec/ld-elf.so.1 $src/bin/xz" -C $out -xf $bootstrapTools
+export LD_LIBRARY_PATH=$out/lib
+
+BADLIST=ld-elf.so.1
+
+oobpatch() {
+    $out/libexec/ld-elf.so.1 $src/bin/cp $1 ./tmp
+    $out/libexec/ld-elf.so.1 $out/bin/patchelf --set-rpath $out/lib --set-interpreter $out/libexec/ld-elf.so.1 ./tmp
+    $out/libexec/ld-elf.so.1 $src/bin/mv ./tmp $1
+    BADLIST="$BADLIST|${1##*/}"
+}
+
+oobpatch $out/bin/patchelf
+oobpatch $out/lib/libthr.so.3
+oobpatch $out/lib/libc.so.7
+
+for f in $($out/libexec/ld-elf.so.1 $out/bin/find $out/lib -type f); do
+    $out/libexec/ld-elf.so.1 $out/bin/grep -E "$BADLIST" <<<"$f" && continue
+    $out/libexec/ld-elf.so.1 $out/bin/patchelf --set-rpath $out/lib $f
+done
+for f in $out/bin/* $out/bin/.*; do
+    $out/libexec/ld-elf.so.1 $out/bin/grep -E "$BADLIST" <<<"$f" &>/dev/null && continue
+    $out/libexec/ld-elf.so.1 $out/bin/patchelf --set-rpath $out/lib --set-interpreter $out/libexec/ld-elf.so.1 $f
+done
+
+unset LD_LIBRARY_PATH
+export PATH=$out/bin
+
+# sanity check
+$out/bin/true || exit 1
+
+# scorched earth
```

Comment:
Could you change this line to something that explains in a sentence what this loop does?

Reply:
done